cst/plugins/eu.esdihumboldt.cst.functions.string/src/eu/esdihumboldt/cst/functions/string/DateExtraction.java

```diff
@@ -65,6 +71,13 @@ protected Object evaluate(String transformationIdentifier, TransformationEngine
 		String sourceString = variables.values().iterator().next().getValueAs(String.class);
 		SimpleDateFormat sdf = new SimpleDateFormat(dateFormat);
 
+		// default ....
+		String leniency = getParameters().get(PARAMETER_LENIENCY).get(0).as(String.class);
+		leniency = getExecutionContext().getVariables().replaceVariables(leniency);
```

Comment:
Only true/false supported, so variable replacement is not really applicable here.